builtin/providers/aws/resource_aws_route53_zone.go

```diff
@@ -104,3 +111,20 @@ func resourceAwsRoute53Wait(r53 *route53.Route53, ref string) (result interface{
 	}
 	return true, status, nil
 }
+
+func resourceAwsGoRoute53Wait(r53 *awsr53.Route53, ref *awsr53.GetChangeRequest) (result interface{}, state string, err error) {
```

Comment:
I had to introduce a separate method specifically for when using `aws-sdk-go` library; `Route53Record` uses the other version here with the `goamz` library. 

This method can replace the original when `Route53Record` is updated.